### DISCUSSION TITLE
Auto-Upgrade to latest TeslaMate software release

### DISCUSSION THREAD

**Discussion by shackrock** [teslamate-org/teslamate#1043] [Ideas]
**What would you like to be added**: I would like to see the software package be able to auto-update to the latest release.

**Why is this needed**: I operate this on a QNAP NAS, and it isa bit intensive to update the docker container. Log in to UI, stop container, SSH in, upgrade, restart the container, hope it works - if not try again.

**Additional context**: For us folks who are somewhat limited on docker/composer experience, this would be a real time saver and hopefully help us not to make any mistakes either!

**Comment by DrMichael** [comment]
Well you could setup a crontab entry with

`docker-compose down; docker-compose pull; docker-compose up -d`

in your teslamate directory.

**Comment by BabyYeggie** [comment]
You can do this with Ourobouros:

   ouroboros:
    container_name: ouroboros
    image: pyouroboros/ouroboros
    environment:
      - CLEANUP=true
      - INTERVAL=604800 ;check interval in seconds (7 days)
      - LOG_LEVEL=info
      - SELF_UPDATE=false
      - IGNORE=postgres mosquitto ;updating postgres to a major release could result in loss of data
    restart: always
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

**Comment by roadrash2108** [comment]
> - IGNORE=postgres mosquitto ;updating postgres to a major release could result in loss of data

Ignoring Postgres is understood, but why not updating Mosquitto? (especially as the explicit version is given in the docker-compose.yml)

**Comment by BabyYeggie** [comment]
> Ignoring Postgres is understood, but why not updating Mosquitto? (especially as the explicit version is given in the docker-compose.yml)

I don't use MQTT, so it's dead to me. Why update something that I don't use?

I also had Traefik in there as well. The update from 2.0 to 2.1 caused an outage for a month. The update from 2.1 to 2.2 caused the entire VM to stop working. I had to backup the data, reset the VM, then re-start from scratch and import the database backup. Luckily, I didn't even notice I was using the wrong docker-compose.yml that allowed Traefik to update from 2.2 to 2.3 since nothing went wrong...
